amethyst_network/src/error.rs

```diff
@@ -0,0 +1,85 @@
+//! Module containing error handling logic.
+
+use std::{
+    fmt::{self, Display, Formatter},
+    io,
+    sync::mpsc,
+};
+
+use crate::server::ServerSocketEvent;
+
+/// The `amethyst_network` result type.
+pub type Error<T> = Result<T, ErrorKind>;
+
+/// Wrapper for all errors who could occur in `amethyst_network`.
+#[derive(Debug)]
+pub enum ErrorKind {
+    /// Error that could occur on the UDP-socket
+    UdpError(laminar::error::NetworkError),
+    /// Error that could occur whit IO.
+    IoError(io::Error),
+    /// Error that could occur when serializing whit `bincode`
+    SerializeError(Box<bincode::ErrorKind>),
```

Comment:
This has an alias, `bincode::Error`, that's probably what we'd want to use.

Reply:
I will take a look at it, because deserialize from bincode returns a boxed error kind. Maybe I can try do dereference it.

Reply:
No need to deref, using the alias directly works (assuming it's available in our version of bincode):
https://docs.rs/bincode/1.0.1/bincode/type.Error.html